RocketChat/Rocket.Chat
# fix: Ignore Empty Spaces in Channel Announcements/Descriptions/Topics fields

## Proposed changes (including videos or screenshots)

This PR addresses the issue where empty spaces in channel announcements/descriptions/topics lead to unexpected behavior. The proposed changes ensured if only empty spaces are present in those field then save button will be disabled  while editing the room information through administration setting or directly from home page

https://github.com/RocketChat/Rocket.Chat/assets/125182653/b67e0ba8-57ca-47b7-b31a-fec7db8b6adc


## Issue(s) 

Closes #31607

## Steps to test or reproduce
1. Click on any channel either through administration setting or directly from home page
2. Go to Room information.
3. Edit the room and add empty spaces to the announcement.
4. Observe the change
